### PR TITLE
[Backport] Use cython<3.0.0 to build oracledb and pyyaml python dependencies

### PR DESCRIPTION
**Pull request by FlorentClarret**
### What does this PR do?

Use cython<3.0.0 to build oracledb and pyyaml python dependencies

### Motivation

Backport of https://github.com/DataDog/datadog-agent/pull/18216

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by cedricvanrompay-datadog — APPROVED** [review]

**Review by vivek-datadog — APPROVED** [review]
👍

File: omnibus/config/software/oracledb-py3.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+       :sha256 => "bb3c391c167b5778ddb15a7538a2b36db5c9b88a50c86c61781ca9ff302bb643",
+       :extract => :seven_zip
+
+relative_path "oracledb-#{version}"
+
+build do
+  license "Apache-2.0"
+  license_file "./LICENSE.txt"
+
+  command "sed -i 's/cython/cython<3.0.0/g' pyproject.toml"
```
Comment (chouetz) [inline omnibus/config/software/oracledb-py3.rb:18]: This won't work for macos (would require `sed -i ""` but I don't know if we need to ensure this

File: omnibus/config/software/pyyaml-py2.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+       :sha256 => "75f966559c5f262dfc44da0f958cc2aa18953ae5021f2c3657b415c5a370045f",
+       :extract => :seven_zip
+
+relative_path "pyyaml-#{version}"
+
+build do
+  license "MIT"
+  license_file "./LICENSE.txt"
+
+  command "sed -i 's/Cython/Cython<3.0.0/g' pyproject.toml"
```
Comment (chouetz) [inline omnibus/config/software/pyyaml-py2.rb:16]: is the capital letter `Cython` expected?

**Review by chouetz — APPROVED** [review]
If you confirm my comments are superfluous LGTM

File: omnibus/config/software/pyyaml-py2.rb (hunk 2 above)
Comment (FlorentClarret) [reply omnibus/config/software/pyyaml-py2.rb:16]: pip is case insensitive, they use Cython and not cython in their own file 🙂

File: omnibus/config/software/oracledb-py3.rb (hunk 1 above)
Comment (FlorentClarret) [reply omnibus/config/software/oracledb-py3.rb:18]: It's ok, I locked it with `if linux?` in the main recipe file

**Pull request merged**